### PR TITLE
Switch publishLocal to use Travis CI workspaces

### PR DESCRIPTION
**Pull request by dwijnand**
https://docs.travis-ci.com/user/using-workspaces/

Caches are meant to be a optimisation only, while workspaces are
intended for sharing files within a build.

The motivation for this change is to address the cache poisoning that
was occuring with cron builds publishing a SNAPSHOT version of Play
while depending on an Akka snapshot version, which was failing the
scripted stage of PR validation because they didn't use Akka's snapshot
resolver.

### CODE REVIEW

**Comment by ignasi35** [comment]
(not a blocker) What do you think about using `workspaces` to share the sources and `target/` too so we don't have to recompile over and over?

**Comment by dwijnand** [comment]
What sources?  With regards to `target/`, that's also worth exploring, but it's not as straight-forward as it might require some data scrubbing (see Hoarder).  Also, I don't know if you can "layer" workspaces.

**Comment by dwijnand** [comment]
An example of using multiple workspaces: https://docs.travis-ci.com/user/using-workspaces/#multiple-workspaces-example

**Comment by ignasi35** [comment]
> What sources? With regards to `target/`, ... 

What I want is to reuse compiled code (aka `target/`). If a new `git clone` is executed on every new VM (but `target/` is reused keeping the original timestamps) won't `sbt` consider the sources to be _newer_ and recompile again? 

IDK, like I said, not a blocker and worth exploring some other time.

**Comment by dwijnand** [comment]
Yeah, the interaction with the timestamps is also a worry.

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,6 @@ install: jabba install "adopt@~1.$TRAVIS_JDK.0-0" && jabba use "$_" && java -Xmx
 stages:
   - validations
   - test
-  - publish-local
```
Comment (mkurz) [inline .travis.yml]: @dwijnand Probably just cosmetics, but why did you remove the `publish-local` stage? But instead merged it into the `validations` stage. Why not just keep `publish-local` but move it just below `validations`?

Comment (dwijnand) [reply .travis.yml]: So that it's not a separate stage (which sounds too much like a circular argument).  Originally I had to separate it from the rest to micro-manage the cache.  Now I can do that at a job level, so I folded it back into `validations`.  The reason to fold it is so it can be run concurrently with `validations` and not be real-time-clock blocker on the next jobs.

Do you have a preference for it being set up some other way?

Comment (mkurz) [reply .travis.yml]: @dwijnand thanks for the explanation, this makes sense to me now. Specially the argument that the publish-local jobs can run concurrently with the validation ones now make a lot of sense :+1:
